### PR TITLE
fix 'show label in DataGrid' option

### PR DESCRIPTION
**Pull request by mikekotikov**

### CODE REVIEW

File: src/components/datagrid/DataGrid.js [hunk 1]
```diff
@@ -458,7 +458,7 @@ export default class DataGridComponent extends NestedArrayComponent {
       const component = this.createComponent(columnComponent, options, row);
       component.parentDisabled = !!this.disabled;
       component.rowIndex = rowIndex;
-      component.inDataGrid = true;
```
Comment (travist) [inline src/components/datagrid/DataGrid.js:461]: We should not be mutating the "component" object.  This is what causes infinite loop errors with the Logic system because the "component" changed. Why is this necessary?

**Review by travist — CHANGES_REQUESTED** [review]

File: src/components/datagrid/DataGrid.js (hunk 1 above)
Comment (mikekotikov) [reply src/components/datagrid/DataGrid.js:461]: The option to 'Show Label in DataGrid' was not available at builder because the custom conditional didn't come true.
`inDataGrid` just wasn't assigned to `editComponent`.
image

Comment (mikekotikov) [reply src/components/datagrid/DataGrid.js:461]: Will find another way to pass it

**Review by MaksimFalei — APPROVED** [review]
